app/models/maintenance_tasks/task_data.rb

```diff
@@ -83,27 +99,23 @@ def code
       File.read(file)
     end
 
-    # Retrieves the latest Run associated with the Task.
+    # Returns the set of currently active Run records associated with the Task.
     #
-    # @return [MaintenanceTasks::Run] the Run record.
-    # @return [nil] if there are no Runs associated with the Task.
-    def last_run
-      return @last_run if defined?(@last_run)
-
-      @last_run = runs.first
+    # @return [ActiveRecord::Relation<MaintenanceTasks::Run>] the relation of
+    #   active Run records.
+    def active_runs
+      @active_runs ||= runs.active
     end
 
-    # Returns the set of Run records associated with the Task previous to the
-    # last Run. This collection represents a historic of past Runs for
-    # information purposes, since the base for Task Data information comes
-    # primarily from the last Run.
+    # Returns the set of completed Run records associated with the Task.
+    # This collection represents a historic of past Runs for information
+    # purposes, since the base for Task Data information comes
+    # primarily from currently active runs.
     #
     # @return [ActiveRecord::Relation<MaintenanceTasks::Run>] the relation of
-    #   record previous to the last Run.
-    def previous_runs
-      return Run.none unless last_run
-
-      runs.where.not(id: last_run.id)
```

Comment:
We used to take the current active on one side, and all the others on the other, now we take active + completed. It means there's some kind of coupling between those two. It may be worth having some kind of test that ensures all valid statuses fall in either one of these buckets. Maybe just `(Run::ACTIVE_STATUSES + Run::COMPLETED_STATUSES).sort == Run::STATUSES.sort` is enough.